tests/unit/test.program.js

```diff
@@ -430,6 +433,203 @@ describe('program.main', () => {
       assert.strictEqual(options.shouldExitProgram, false);
     });
   });
+
+  it('applies config from config file when specified', () => {
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const fakePath = 'path/to/web-ext-config.js';
+    const resolvedFakePath = path.resolve(fakePath);
+    const fakeLoadJSConfigFile = sinon.spy(() => {});
+    const fakeApplyConfigToArgv = sinon.spy(() => {});
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(() => {
+        const options = fakeCommands.lint.firstCall.args[0];
+        assert.strictEqual(options.config, fakePath);
+        sinon.assert.calledOnce(fakeLoadJSConfigFile);
+        sinon.assert.calledOnce(fakeApplyConfigToArgv);
+        sinon.assert.calledWith(fakeLoadJSConfigFile,
+          sinon.match(resolvedFakePath));
+        sinon.assert.calledWith(fakeApplyConfigToArgv, sinon.match({
+          configFileName: resolvedFakePath,
+        }));
+      });
+  });
+
+  it('logs and throws when config file was not loaded', () => {
+    const fakeProcess = createFakeProcess();
+    const logger = fake(createLogger(__filename));
+    const fakeCommands = fake(commands, {
+      lint: () => Promise.resolve(),
+    });
+    const fakePath = 'path/to/web-ext-config.js';
+    const fakeLoadJSConfigFile = sinon.spy(() => {
+      throw new UsageError('bad path');
+    });
+    const fakeApplyConfigToArgv = sinon.spy(() => {});
+
+    return execProgram(
+      ['lint', '--config', fakePath],
+      {
+        commands: fakeCommands,
+        runOptions: {
+          shouldExitProgram: false,
+          systemProcess: fakeProcess,
+          logger,
+          applyConfigToArgv: fakeApplyConfigToArgv,
+          loadJSConfigFile: fakeLoadJSConfigFile,
+        },
+      })
+      .then(makeSureItFails())
+      .catch(onlyInstancesOf(UsageError, (error) => {
+        assert.match(error.message, /bad path/);
+        sinon.assert.notCalled(fakeApplyConfigToArgv);
+        sinon.assert.notCalled(fakeProcess.exit);
+        sinon.assert.calledOnce(logger.error);
+      }));
+  });
+
+  it('throws when config file loading fails', () => {
```

Comment:
to make it a bit more clear which is the difference between this test and the previous one,
we could rephrase this description to "throws when config file loading raises an unexpected error" 
(given that we only expect it to raise a `UsageError` in the known error cases)